src/Compilers/CSharp/Portable/Binder/Semantics/BestTypeInferrer.cs

```diff
@@ -171,22 +176,31 @@ private TypeSymbol GetBestType(ImmutableArray<TypeSymbol> types, ref HashSet<Dia
             // that every type *before* best was also worse.
             for (int i = 0; i < bestIndex; i++)
             {
-                TypeSymbol type = types[i];
-                TypeSymbol better = Better(best, type, ref useSiteDiagnostics);
+                var type = types[i];
+                var better = Better(best, type, ref useSiteDiagnostics);
 
-                if (better != best)
+                if (!best.Equals(better, TypeCompareKind.ConsiderEverything))
                 {
                     return null;
                 }
             }
 
+            // If any of the types are null, the result should be nullable.
+            if (includeNullability &&
```

Comment:
Why? Just because we don't know the type, doesn't mean that the value can be null. e.g. if the inputs are a delegate value and a lambda expression.

Reply:
We do know the type in this case, it is `bestType`.